websub-ballerina/http_service.bal

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+import ballerina/log;
+import ballerina/jballerina.java;
+
+service class HttpService {
+    private SubscriberService subscriberService;
+    private SubscriberServiceConfiguration serviceConfig;
+    private string callbackUrl;
+    private boolean isSubscriptionValidationDeniedAvailable = false;
+    private boolean isSubscriptionVerificationAvailable = false;
+    private boolean isEventNotificationAvailable = false;
+
+    # Invoked during the initialization of a `websub:HttpService`
+    #
+    # + subscriberService   - {@code websub:SubscriberService} provided service
+    # + serviceConfig       - {@code SubscriberServiceConfiguration} subscriber-service
+    #                          related configurations
+    # + callback            - {@code string} dynamically generated callback-url
+    public function init(SubscriberService subscriberService, SubscriberServiceConfiguration config, 
+                         string callback) returns error? {
+        self.subscriberService = subscriberService;
+        self.serviceConfig = config;
+        self.callbackUrl = callback;
+        
+        string[] methodNames = getServiceMethodNames(subscriberService);
+        
+        foreach var methodName in methodNames {
+            match methodName {
+                "onSubscriptionValidationDenied" => {
+                    self.isSubscriptionValidationDeniedAvailable = true;
+                }
+                "onSubscriptionVerification" => {
+                    self.isSubscriptionVerificationAvailable = true;
+                }
+                "onEventNotification" => {
+                    self.isEventNotificationAvailable = true;
+                }
+                _ => {}
```

Comment:
I think it is better log something here. Maybe an error log ?

Reply:
* Updated with necessary logs.